src/benchmark_runner.cc

```diff
@@ -257,6 +261,25 @@ BenchmarkRunner::BenchmarkRunner(
   }
 }
 
+void BenchmarkRunner::UpdateReport(RunResults& run_results) {
```

Comment:
i don't understand why this is necessary. can you add a comment explaining it please?

Reply:
The BenchmarkName from the RunResult object was created based on the BenchmarkInstance's BenchmarkName.
Hence its min_time could be zero (if the user never calls `Benchmark::MinTime(...))`). 

We want to update that to the value set in the command-line flag, if user specified the flag.

The logic in this is also a bit convoluted:
 - The default value of BenchmarkInstance's min_time is 0.0 
 - The default value of command-line flag min_time is 0.5
 - We update the name to BenchmarkRunner's min_time only if it's neither of the default values. (*)

(*) This is still not very good reasoning. Maybe we want a way to really know whether user overrides the default value or not and only print the value in that case. (If they override the default value with a default, that's fine too - we'd still print it. Eg., if they set --benchmark_min_time=0.0, we should still print it)

WDYT?

Reply:
i think we should be more consistent. whether the benchmark min time was set by the macro or by the commandline flag, it should be captured somewhere that doesn't require posthoc rewriting of history.

ie, what happens if we store `b.min_time` from the macro _or_ the flag? then all the benchmark naming should just work, right?

Reply:
also, do we want to add a suffix for iterations if the user overrode the iterations on the command line?

Reply:
Ooh, I see - hmm. I've only added this part so that we can test the flag's setting 
(since there's no way to pull the min_time field out of BenchmarkRunner from a test) ...

any suggestion?

Otherwise, maybe can just rewrite the test to test that the flag was parsed correctly?

Reply:
maybe just add the min time to the benchmark runner so we can access it for testing? :)

Reply:
Done. 

(ended up added a hook to report the config from BenchmarkRunner to the BenchmarkReporter interface, because from a test, we dont have access to BenchmarkRunner objects)